pkg/datasource/accessors.go

```diff
@@ -379,6 +388,37 @@ func (a *fieldAccessor) Flags() uint32 {
 	return a.f.Flags
 }
 
+func (a *fieldAccessor) Tags() []string {
+	return slices.Clone(a.f.Tags)
+}
+
+func (a *fieldAccessor) HasAllTagsOf(tags ...string) bool {
+	for _, t1 := range tags {
+		found := false
+		for _, t2 := range a.f.Tags {
+			if t1 == t2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Here we go again with them slices, haha. I looked at it earlier and thought "didn't I change it? No, it has two iterators, I did it somewhere else" but didn't connect the dots that it can be used for the inner loop. Nice!